pkg/app/config.go

```diff
@@ -510,7 +524,16 @@ func MergeAzureUserArgs(config Config, azureArgs *AzureArgs) error {
 		return fmt.Errorf("failed to set %s: %w", PackerAzureGalleryImagePublisherPath, err)
 	}
 
-	if err := config.Set(PackerAzureGalleryImageSKU, azureArgs.GalleryImageSKU); err != nil {
+	galleryImageSKU := azureArgs.GalleryImageSKU
+	if galleryImageSKU == "" {
```

Comment:
would it be a terrible thing to require this instead of setting it to the gallery name?

Reply:
The earlier implementation defaults to `distribution_version` if its not provided. If we make it `required` we have to make sure to validate and fail earlier. The validation would require call to Azure and make sure the SKU does not exists within given `subscription/resourcegroup/gallery/gallery_image_id`. 
IMHO, we should remove/deprecate this flag in the CLI and not ask gallery_name, sku etc to simplify provision. 
The azure documentation around gallery offer, publisher and sku and gallery image URN are not clear enough. Instead we can make sure that KIB ensures unique URN for the image.